codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/ResolveClientConfig.java

```diff
@@ -0,0 +1,96 @@
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.List;
+import java.util.logging.Logger;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoDelegator;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.SmithyGoDependency;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.utils.ListUtils;
+
+/**
+ * Registers additional client specific configuration fields
+ */
+public class ResolveClientConfig implements GoIntegration {
```

Comment:
This class should probably be named specific to S3 since the option being added are specific to those services. Also since the internals of this class aren't really generic, and pretty specific to the S3 options.

Reply:
We can extend the class easily to be used by other services. I would want to leave the naming as is.  

It seems tightly coupled as we have predicates written for s3shared at the moment.